### ISSUE TITLE
"partial" is inconsistent in C#

### ISSUE THREAD

**Issue by schani** [quicktype/quicktype#601]
A user requested adding `partial` annotations even for "just types", because they want to extend their types with more stuff that's not in the JSON.  Looking at this further, I discovered that when we generate helpers, we make ALL classes partial, despite only the top-level ones being extended by the helpers.

We need to decide whether

* `partial` is there to let users extend the types.  Then it should be there even for "just types".

* `partial` is just there for the helpers.  Then we need to remove it from all the non-top-level classes.

I think I weakly prefer the former.  If we go with the latter I think we should at least have the option to make all classes partial.

**Comment by silverkea** [comment]
Thanks for considering this. Nice job with quicktype too!

With generated code I often find it's useful to be able to add some additional properties that aren't in the schema but may be used internally by code. Ideally I like to be able to keep my generated code in separate files to the additional custom code, so any time the schema is updated the generated code can be replaced without loosing any additional properties etc.

For instance, at the moment I'm working on JSON for a report that has detail items and a summary calculated from the detail items. To calculate the summary I need some fields on the detail items that, for business reasons, I don't want to expose in the schema for the detail items. 

Ideally I would like to be able to do a query for my detail items, and project the result into the detail objects for the report (including the additional properties), then calculate the summary using the additional properties on those objects, and then serialize to json with the additional properties being excluded because the I have JsonIgnore attributes in the custom partial classes.

I can do this at the moment using the "Complete" generation option for C#, but for our solution we don't really use the serialization helper classes, so I would ideally like to be able to generate partial classes for the "Attributes Only" and "Just Types" options.

Example:
```
//Foo.generated.cs

public partial class Foo
{
    [JsonProperty("bar")]    
    public string Bar { get; set;}
}

//Foo.custom.cs

public partial class Foo 
{
    [JsonIgnore]
    public string SomethingExtraNotInSchema { get; set; }
}
```

**Comment by schani** [comment]
@dvdsgl Do you have strong opinions on this?

**Comment by dvdsgl** [comment]
I prefer the latter as well–all generated classes should be partial in C#.

At first I'd say that classes like `Converter` and `Serialize` should *not* be partial, but this could be a subtle way to avoid the collision problem in #276 (provided that we sufficiently distinguish member names).

**Comment by schani** [comment]
Fixed in #602
